sip/sip-003-peer-network.md

```diff
@@ -764,6 +791,34 @@ again.
 When executing a handshake, a peer should _not_ include any other peers in the
 `relayers` vector except for itself.  The `relayers` field will be ignored.
 
+### Learning the public IP address
+
+Before the peer can participate in the control plane, it must know its
+publicly-routable IP address so it can exchange it with its remote neighbors.
+This is necessary, since other neighbors-of-neighbors will learn this peer's
+public IP address from its remote neighbors, and thus must have a publicly-routable
+address if they are going to handshake with it.
+
+The peer may take an operator-given public IP address.  If no public IP address
+is given, the peer will learn the IP address using the `NatPunchRequest` and
+`NatPunchReply` messages as follows:
+
+1. The peer sends a `NatPunchRequest` to a randomly-chosen neighbor it has
+   already handshaked with.  It uses a random nonce value.
+2. The remote neighbor replies with a (signed) `NatPunchReply` message, with its
+   `addrbytes` and `port` set to what it believes the public IP is (based on the
+   underlying socket's peer address).
```

Comment:
Can remote neighbour reply with an IP that they control and become a MITM?

Reply:
The remote neighbor can reply with an arbitrary IP.  However, steps 3 and 4 prevent the peer from believing that this IP is its public IP, because in steps 3 and 4, the local peer attempts to connect to itself with the IP given by the remote peer.